src/FFI.php

```diff
@@ -429,6 +432,59 @@ private static function init(): void
     int connect_flags);
 
 const char* g_param_spec_get_blurb (GParamSpec* psp);
+
+typedef struct _GClosure GClosure;
+typedef void (*marshaler)(
+    struct GClosure* closure, 
+    GValue* return_value, 
+    int n_param_values, 
+    const GValue* param_values, 
+    void* invocation_hint, 
+    void* marshal_data
+);
+
+typedef struct _GClosureNotifyData GClosureNotifyData;
+struct _GClosureNotifyData
+{
+  void*       data;
+  GClosureNotify notify;
+};
+struct _GClosure
+{
+  /*< private >*/
+  int ref_count : 15;  /* (atomic) */
```

Comment:
Oooof this is ugly, do we really need to declare all these fields? I think `GClosure` should just be an opaque pointer, shouldn't it? We don't need to use sizeof, or fetch any members. Like `GObject` and `GTypeInstance` we can just omit most or even all fields.

Reply:
Ah wait, we do need `g_closure_new_simple(\FFI::sizeof(FFI::ctypes('GClosure')), null);`. How about just having a constant for `sizeof(GClosure)`? It'd be much safer than declaring all the members and hoping that php-ffi implements features like bitfield packing correctly.

Reply:
Ah you set `->marshal` as well. Help, this feels really bad. Maybe there's another approach here? I'll try some experiments.